onyx-database/src/com/onyx/fetch/impl/IndexScanner.java

```diff
@@ -56,12 +57,31 @@ public IndexScanner(QueryCriteria criteria, Class classToScan, EntityDescriptor
                 if(query.isTerminated())
                     return returnValue;
 
-                indexController.findAll(idValue).forEach(o -> references.add(o));
+                if(criteria.getOperator() == QueryCriteriaOperator.GREATER_THAN)
```

Comment:
should be: 
` QueryCriteriaOperator.GREATER_THAN.equals(critiera.getOperator())`